src/FlagLocalization.cpp

```diff
@@ -0,0 +1,206 @@
+#include <string>
+#include <unistd.h>
+#include <iostream>
+#include "ros/ros.h"
+#include "std_msgs/Float64.h"
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <std_msgs/Float32.h>
+#include "sensor_msgs/LaserScan.h"
+#include "sensor_msgs/PointCloud.h"
+#include "geometry_msgs/PointStamped.h"
+#include "laser_geometry/laser_geometry.h"
+#include <tf2_ros/transform_listener.h>
+#include <geometry_msgs/TransformStamped.h>
+#include <math.h>
+#include <tf/transform_broadcaster.h>
+
+#define PI 3.14159265
+
+using namespace std;
+
+class Listener
+{
+public:
+	void scanCB(const sensor_msgs::LaserScan::ConstPtr& scan_in);
+	geometry_msgs::PointStamped getPoint1() { return _avgPt_1; }			//Average points found by adding all x and y values for each flag then dividing
+	geometry_msgs::PointStamped getPoint2() { return _avgPt_2; }			//by the # of points.  These happen to be geometric centroids of semicircles seen
+																		//by the lidar.
+
+private:
+	laser_geometry::LaserProjection projector_;
+	geometry_msgs::PointStamped _avgPt_1;
+	geometry_msgs::PointStamped _avgPt_2;
+
+};
+
+float pnt1x, pnt2x, pnt1y, pnt2y;
+
+void Listener::scanCB (const sensor_msgs::LaserScan::ConstPtr& scan_in)
+{
+	sensor_msgs::PointCloud cloud;
+	projector_.projectLaser(*scan_in, cloud);
+
+	//Radius of flags in meters
+	float rad1 = .1;
+	float rad2 = .04;
+	//Value to determine tolerance between the two flags, 
+	//should be a little less than half of the distance between flags,
+	//but greater than the diameter of large flag
+	float gap = .05;
+
+	//Variables used by code
+		int n = cloud.points.size();		// Number of points total
+		
+		float xsum1 = 0;			//Used to find an average point for the first cluster of points(flag 1)
+		float ysum1 = 0;
+
+		float xsum2 = 0;			//Used to find an average point for the first cluster of points(flag 2)
+		float ysum2 = 0;
+
+		int pts1 = 0;				//pts1 and 2 are keeping track of the number of points collected for each flag
+		int pts2 = 0;				//These are used in calculating the avg points
+
+		float theta1;				//Angles from beacon to average points.  By extending along this line,
+		float theta2;				//a center of each flag can be calculated.
+
+		int obj = 1;				//The variable obj is used to distinguish whether a point belongs to the first flag or second
+
+		float yaw;				//This is the yaw of the output frame "flag" and used creating the transform between "beacon_frame" and "flag"
```

Comment:
Fix indentation

Reply:
What indentation did you want fixed here?

Reply:
I tabbed it in one extra to show that these variables aren't to be edited like the ones above them if that's what the issue is.

Reply:
Yeah that's non-standard

Reply:
Indentation should follow blocks of code { } or wrapping one statement to a new line